docs/readmes/proposals/p005_call_tracing.md

```diff
@@ -0,0 +1,283 @@
+    required:
+      - type
+    properties:
+      trace_type:
+        type: string
+        x-nullable: false
+        enum:
+          - 'GATEWAY'
+          - 'SUBSCRIBER'
+          - 'PROTOCOL'
+          - 'INTERFACE'
+      imsi:
+        type: string
+      protocol:
+        type: string
+      interface:
+        type: string
+      timeout:
+        type: integer
+        format: uint32
+        description: Timeout of call trace in seconds
+
+  call_trace state:
+    type: object
+    description: Full state object of a call trace
+    properties:
+      call_trace_available:
+        type: boolean
+```
+
+#### Orchestrator - ctraced
+
+On the gateway, the ctraced service will manage call tracing for the network.
+This service will be responsible for initiating call tracing sessions on
+gateways, and servicing downloads of call trace captures.
+
+Call traces will be stored in postgresql, and periodically, ctraced should
+delete old traces.
+
+#### Orchestrator - Trace storage
+
+Storage of call trace captures will be done in postgresql, stored as binary
+data. A new table will be created for storage of call traces.
+
+A migration script is required to ensure that existing deployments will get
+this new table for storing call traces.
+
+#### Gateway - ctraced
+
+The ctraced service on the gateway will be a new service, responsible for
+managing call tracing. Options will be provided for gateway wide
+packet-capture, and later on, options will be provided for protocol, interface,
+or subscriber specific call tracing.
+
+```
+service CallTraceService {
+  rpc StartCallTrace (StartTraceRequest) returns (StartTraceResponse) {}
+
+  rpc EndCallTrace (EndTraceRequest) returns (EndTraceResponse) {}
+}
+
+
+message StartTraceRequest {
+  enum TraceType {
+    ALL = 0;
+    SUBSCRIBER = 1;
+    PROTOCOL = 2;
+    INTERFACE = 3;
+  }
+
+  TraceType trace_type = 1;
+  // IMSI specified only if trace_type is SUBSCRIBER
+  string imsi = 2; // Include prefix 'IMSI'
+
+  enum ProtocolName {
+    SCTP = 0;
+    DIAMETER = 1;
+  }
+  // Protocol name specified only if trace_type is PROTOCOL
+  ProtocolName protocol = 3;
+  enum InterfaceName {
+    S1AP = 0;
+    GX = 1;
+    GT = 2;
+  }
+  // Interface name specified only if trace_type is INTERFACE
+  InterfaceName interface = 4;
+}
+
+message StartTraceResponse {
+  bool success = 1; // May fail due to an existing tracing session
+}
+
+message EndTraceRequest {}
+
+message EndTraceResponse {
+  bool success = 1; // May fail due to no existing tracing session
+  bytes trace_content = 1; // Max size of 4MB
+}
+```
+
+To perform call tracing, ctraced will use `tshark`, a terminal-based wireshark
+tool to do packet capture on the gateway.
+
+Call tracing will be initiated from the orc8r, and the packet-capture will be
+transferred back to orc8r from AGW via gRPC. Currently there is a limit for
+gRPC message sizes to 4MB, which limits the size of packet capture. Given that
```

Comment:
Nice to know. Yes. I think setting a max capture size of 1-2MB should mostly suffice.